extract_timeseries_tar.py

```diff
@@ -18,68 +18,162 @@
 import bids
 
 ATLAS_METADATA = {
-    'difumo': {'type': "dynamic",
-               'dimensions': [64, 128, 256, 512, 1024],
-               'resolutions': [2, 3],
-               'label_idx': 1,
-               'fetcher': "nilearn.datasets.fetch_atlas_difumo(dimension={dimension}, resolution_mm={resolution}, data_dir=\"{atlas_path}\")"},
-    'segmented_difumo': {'type': "dynamic",
-                         'dimensions': [64, 128, 256, 512, 1024],
-                         'resolutions': [2, 3],
-                         'fetcher': "segmented_difumo_fetcher(dimension={dimension}, resolution_mm={resolution}, atlas_path=\"{atlas_path}\")"},
-}
-
-#TODO: mask data using subject mask
-#TODO: wait you PR before making a standalone tool (templateflow downloading, confound parameters loading)
-#TODO: QC timeseries https://github.com/SIMEXP/mapsmasker_benchmark/blob/main/mapsmasker_benchmark/main.py 
-def segmented_difumo_fetcher(atlas_path, dimension=64, resolution_mm=3):
-
-    templateflow.conf.TF_HOME = pathlib.Path(atlas_path)
-    templateflow.conf.init_layout()
-    templateflow.conf.update(local=True)
-
-    img_path = str(templateflow.api.get("MNI152NLin2009cAsym", atlas="DiFuMo",
-                                        desc=f"{dimension}dimensionsSegmented", resolution=f"0{resolution_mm}", extension=".nii.gz"))
-    label_path = str(templateflow.api.get("MNI152NLin2009cAsym", atlas="DiFuMo",
-                                          desc=f"{dimension}dimensionsSegmented", resolution=f"0{resolution_mm}", extension=".tsv"))
+    # 'schaefer': {
+    #     'source': "templateflow",
+    #     'templates' : ['MNI152NLin2009cAsym', 'MNI152NLin6Asym'],
+    #     'resolutions': ["01", "02"],
+    #     'atlas': 'Schaefer2018',
+    #     'description_pattern': "{dimension}Parcels{network}Networks",
+    #     'dimensions': [100, 200, 300, 400, 500, 600, 800, 1000],
+    #     'networks': [7, 17],
+    #     'atlas_parameters': ['resolution', 'description'],
+    #     'label_parameters': ['description'],
+    #     },
+    'schaefer7': {
+        'source': "templateflow",
+        'templates' : ['MNI152NLin2009cAsym', 'MNI152NLin6Asym'],
+        'resolutions': ["01", "02"],
+        'atlas': 'Schaefer2018',
+        'description_pattern': "{dimension}Parcels7Networks",
+        'dimensions': [100, 200, 300, 400, 500, 600, 800, 1000],
+        'atlas_parameters': ['resolution', 'description'],
+        'label_parameters': ['description'],
```

Comment:
So I have trouble understanding what you wanted to do there with the atlas_parameters and label_parameters attributes ecause now it is unused in all the code?
Should it be inside `generate_templateflow_parameters` ?

Reply:
This object should be static, and not meant to be changed at runtime.
I think what you wanted here is to get a templateflow like object. So for this you will have no choice to create another object that would be instanciated at runtime:
```
CURR_ATLAS: {'name': Schaefer2018, 'dimensions': 100, 'description_pattern': "{dimension}Parcels7Networks", etc..}
```
This will greatly improve readability (and not mix static VS dynamic objects)

Reply:
Also not sure if this was intended but `schaefer7` does not have `'resolution'` for `'label_parameters'` key

Reply:
> Also not sure if this was intended but `schaefer7` does not have `'resolution'` for `'label_parameters'` key

Yes this is intentional. If you check the original atlas, the label file doesn't have resolution related information. This is why I have `atlas_parameters` and `label_parameters` as separated things

Reply:
> The `ATLAS_METADATA` object is static, and not meant to be changed at runtime.

You are absolutely right. The function should make a copy and work on the copy.

Reply:
This is now fixed